### DISCUSSION TITLE
[feature] shortest_path described as the sequence of edges (multigraph)

### DISCUSSION THREAD

**Discussion by kkdd** [networkx/networkx#5818] [Q&A] [answered]
Hello,
Can we get the sequence of edges of shortest_path?
I think that a path should not be described as the sequence of nodes but that of edges when multigraph.

**Comment by dschult** [comment, marked as answer]
We return nodes because edges are easy to form from the nodes:
```edges = nx.utils.pairwise(nodes)```
The case of MultiGraph is slightly more difficult because you have to figure out which edge the path took between nodes. But that is simply a matter of finding the shortest edge between each node:
```python
multiedges = [(u, v, min(G[u][v], key=lambda k: G[u][v][k].get(weight, 1))) for u, v in nx.utils.pairwise(nodes)]
```
(not tested)

**Reply by kkdd** [reply]
Thank you for your advice.
It works:

```
import networkx as nx
def edges_from_shortest_path(G, nodes):
  edges = nx.utils.pairwise(nodes)
  if G.__class__.__name__ == 'Graph' or G.__class__.__name__ == 'DiGraph':
  	return edges
  return [(u, v, min(G[u][v], key=lambda k: G[u][v][k].get('weight', 1))) for u, v in edges]

MG = nx.MultiDiGraph()
MG.add_weighted_edges_from([(1, 2, 0.75), (1, 2, 0.5), (2, 3, 0.5), (1, 3, 1.5)])
print(edges_from_shortest_path(MG, nx.dijkstra_path(MG, 1, 3)))
[(1, 2, 1), (2, 3, 0)]
```

**Reply by rossbar** [reply]
Thanks @dschult for the concise explanation. It might be worth adding this (or a more concise version of) this example to the docstring.

**Reply by Qudirah** [reply]
@rossbar 

The documentation will be added to what folder?

**Reply by navyagarwal** [reply]
@Qudirah I think it'll be added to the docstring of dijkstra_path method under `algorithms/shortest_paths/weighted.py`

**Reply by navyagarwal** [reply]
@rossbar Created a PR for this over [here](https://github.com/networkx/networkx/pull/6497)

**Reply by Qudirah** [reply]
> @rossbar Created a PR for this over [here](https://github.com/networkx/networkx/pull/6497)

:( was hoping to do that. Good work anyways.

**Reply by navyagarwal** [reply]
> :( was hoping to do that. Good work anyways.

Oh, I'm really sorry about that :(
